src/hazelcore/semantics/CursorInfo.re

```diff
@@ -202,7 +216,11 @@ let staging_left_border = ci =>
       Exp(
         Inj(_, _, _) | Parenthesized(_) | Case(_, _, _, _) | Lam(_, _, _, _),
       ) |
-      Pat(Inj(_, _, _) | Parenthesized(_)) |
+      Pat(
+        VarPat(Inj(_, _, _), _) | VarPat(Parenthesized(_), _) |
```

Comment:
(the change I suggested above makes it so we don't need to consider these invalid cases)